### PR TITLE
test: adding tests for jsobject regression tests

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
/ok-to-test tags="@tag.All"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12542379469>
> Commit: eafb3a10c0f5b71cc6162ed4f90f5ed3a7adfbd4
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12542379469&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Mon, 30 Dec 2024 07:54:20 UTC

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive test suite for JavaScript objects, validating functionalities such as creation, deletion, and user interface interactions.
	- Added a new test case for forking applications, ensuring successful operation and validation of JavaScript objects.

- **Improvements**
	- Enhanced the `EntityExplorer`, `JSEditor`, and `CommonLocators` classes with new properties for better management of entities and actions.
	- Made the `_forkModal` property in the `HomePage` class publicly accessible.

- **Bug Fixes**
	- Addressed issues with JSObjects having identical names and ensured proper alert messages are displayed.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    agHelper.AssertContains("JSObject2", "exist", entityExplorer._entityName);
+
+    // Delete JSObj
+    agHelper.GetNClick(jsEditor._addJSObj, 0, true);
+    agHelper.GetNClick(jsEditor._moreActions, 0, true);
+    agHelper.GetNClickByContains(HomePage.portalMenuItem, "Delete");
+    agHelper.GetNClickByContains(HomePage.portalMenuItem, "Are you sure?");
+    agHelper.ValidateToastMessage("JSObject4 deleted successfully");
+
+    // Verify Copy to Page
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts:70]: Please give bug link.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Comment by NandanAnantharamu** [comment]
/ci-test-limit-count run_count=1 update_snapshot=false specs_to_run=cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts,
cypress/e2e/Regression/ClientSide/JSObject/JSObject_ForkApp_spec.ts

**Comment by sagar-qa007** [comment]
/ci-test-limit-count run_count=1 update_snapshot=true specs_to_run=cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts,
cypress/e2e/Regression/ClientSide/JSObject/JSObject_ForkApp_spec.ts runId=12460935225

**Comment by sagar-qa007** [comment]
/ci-test-limit-count run_count=1 update_snapshot=true specs_to_run=cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts,cypress/e2e/Regression/ClientSide/JSObject/JSObject_ForkApp_spec.ts runId=12460935225

**Comment by NandanAnantharamu** [comment]
/ci-test-limit-count run_count=10 update_snapshot=true specs_to_run=cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts,cypress/e2e/Regression/ClientSide/JSObject/JSObject_ForkApp_spec.ts runId=12460935225

File: app/client/cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/JSObject/JSObject_Tests_spec.ts]: this is not a bug hence no bug link added, also this part of the code is covered at many places hence removing this redundant block

**Review by sagar-qa007 — DISMISSED** [review]

**Review by sagar-qa007 — DISMISSED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**